### PR TITLE
Require Node.js 12.2, rework `.fn` to `pProgress` and export `PProgress` as a named export

### LINKED ISSUE #18: Rework API to target async/await interfaces
Instead of doing this:

```js
const progressPromise = new PProgress((resolve, reject, progress) => {
	const job = new Job();

	job.on('data', data => {
		progress(data.length / job.totalSize);
	});

	job.on('finish', resolve);
	job.on('error', reject);
});
```

The API could instead be reworked to do this:

```js
const progressPromise = pProgress(async progress => {
	const job = new Job();

	job.on('data', data => {
		progress(data.length / job.totalSize);
	});

	job.on('error', error => {
		throw error
	});

	return await pEvent(job, 'finish')
})
```

With the legacy `Promise` interface working like this:

```js
const progressPromise = pProgress(progress => new Promise((resolve, reject) => {
	const job = new Job();

	job.on('data', data => {
		progress(data.length / job.totalSize);
	});

	job.on('finish', resolve);
	job.on('error', reject);
}))
```

It looks like `PProgress.fn` already does something similar but you still have to invoke the returned function:

https://github.com/sindresorhus/p-progress/blob/e63f3becd8f4d83e28b868920d1e3084b7091344/index.js#L36-L65

### PR DESCRIPTION
**Pull request by Richienb**
Fixes #18, Fixes #15

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -13,35 +13,64 @@ $ npm install p-progress
 ## Usage
 
 ```js
-const PProgress = require('p-progress');
+import {PProgress} from 'p-progress';
```
Comment (sindresorhus) [inline readme.md]: The main example should be the `pProgress` default export. This example can be in the API docs for `new PProgress`.

**Comment by sindresorhus** [comment]
Any thoughts on exporting the `.all` method as `pProgessAll` instead?

**Comment by Richienb** [comment]
`PProgress` extends `Promise` so removing the override of `.all` wouldn't remove the function. Re-exporting it though might be useful.

**Comment by sindresorhus** [comment]
Ahh, good point. I didn't think of that. Then we can just keep it as is.

**Pull request merged**